lib/src/detectors/vmware_detector.cc

```diff
@@ -0,0 +1,57 @@
+#include <internal/detectors/vmware_detector.hpp>
+#include <internal/vm.hpp>
+#include <leatherman/execution/execution.hpp>
+#include <leatherman/util/regex.hpp>
+
+using namespace whereami;
+using namespace leatherman::util;
+using namespace std;
+
+namespace whereami { namespace detectors {
+
+    result vmware(const sources::cpuid_base& cpuid_source,
+                  const sources::dmi_base& dmi_source)
+    {
+        result res {vm::vmware};
+
+        if (dmi_source.manufacturer() == "VMware, Inc." ||
+            cpuid_source.vendor() == "VMwareVMware") {
+            res.validate();
+
+            auto bios_address = dmi_source.bios_address();
+            if (!bios_address.empty()) {
+                auto value = stoi(bios_address, nullptr, 16);
+                res.set("version", vmware_bios_address_to_version(value));
+            }
+        }
+
+        return res;
+    }
+
+    string vmware_bios_address_to_version(int address)
+    {
+        static const std::unordered_map<int, std::string> version_map {
```

Comment:
Seems brittle, but we have plenty of other examples of similar maps that require manual attention. Probably not much else we can do here.